plugin/src/main/java/io/fabric8/maven/plugin/HelmMojo.java

```
@@ -0,0 +1,248 @@
+            throw new MojoExecutionException("No Chart name defined! Please specify the `" + PROPERTY_HELM_CHART_NAME + "` property");
+        }
+        generateHelmChartDirectory(this.kubernetesSourceDir, this.kubernetesOutputDir, "helm");
+        generateHelmChartDirectory(this.openshiftSourceDir, this.openshiftOutputDir, "helm-openshift");
+    }
+
+    protected void generateHelmChartDirectory(File sourceDir, File outputDir, String classifier) throws MojoExecutionException {
+        getLog().info("Creating Helm Chart " + chartName + " in " + outputDir + " for manifest folder : " + sourceDir);
+
+        if (Files.isDirectory(outputDir)) {
+            Files.recursiveDelete(outputDir);
+        }
+        if (!sourceDir.isDirectory() || !sourceDir.exists()) {
+            getLog().warn("Chart source directory " + sourceDir + " does not exist so cannot make chart " + chartName);
+            return;
+        }
+        if (!containsYamlFiles(sourceDir)) {
+            getLog().warn("Chart source directory " + sourceDir + " does not contain any YAML manifests or templates so cannot make chart " + chartName);
+            return;
+        }
+        File templatesDir = new File(outputDir, "templates");
+        templatesDir.mkdirs();
+        try {
+            Files.copy(sourceDir, templatesDir);
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to copy manifest files from " + sourceDir + " to chart templates directory: " + templatesDir + ". Reason: " + e, e);
+        }
+
+        File outputChartFile = new File(outputDir, "Chart" + HELM_YAML_EXTENSION);
+        Chart chart = createChart();
+        try {
+            KubernetesHelper.saveYaml(chart, outputChartFile);
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to save chart " + outputChartFile + ". Reason: " + e, e);
+        }
+
+        File basedir = null;
+        if (project != null) {
```

will never be null. And if so, an NPE would be better than silently ignoring it. I just would leave out the check.